src/main/java/com/github/cloudyrock/mongock/Mongock.java

```diff
@@ -107,7 +109,19 @@ void execute() throws MongockException {
       lockChecker.releaseLockDefault();//we do it anyway, it's idempotent
       logger.info("Mongock has finished his job.");
     }
+  }
 
+  //this is to force sonar issue
+  Object ex;
+  private Object getException() {
+    if(ex == null) {
+      synchronized (this) {
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Remove this dangerous instance of double-checked locking. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2168)